app/src/main/java/com/google/droidjump/LevelsFragment.java

```diff
@@ -29,20 +27,22 @@
 import android.widget.ImageButton;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
 import com.google.droidjump.models.LevelManager;
+import com.google.droidjump.models.NavigationHelper;
 import java.util.Objects;
 
 /**
  * Displays Levels Screen.
  */
 public class LevelsFragment extends Fragment {
-    private MainActivity activity;
+    private FragmentActivity activity;
     private LevelsAdapter adapter;
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        activity = Objects.requireNonNull((MainActivity) getActivity());
+        activity = getActivity();
```

Comment:
maybe cast it here to MainActivity

Reply:
Do not think that we need to cast it in case we are using parent class FragmentActivity.